src/realm/sync/client.cpp

```diff
@@ -534,26 +546,28 @@ void ClientImpl::register_unactualized_session_wrapper(SessionWrapper* wrapper,
 void ClientImpl::register_abandoned_session_wrapper(util::bind_ptr<SessionWrapper> wrapper) noexcept
 {
     // Thread safety required.
-
-    std::lock_guard lock{m_mutex};
-    REALM_ASSERT(m_actualize_and_finalize);
-
-    // If the session wrapper has not yet been actualized (on the event loop
-    // thread), it can be immediately finalized. This ensures that we will
-    // generally not actualize a session wrapper that has already been
-    // abandoned.
-    auto i = m_unactualized_session_wrappers.find(wrapper.get());
-    if (i != m_unactualized_session_wrappers.end()) {
-        m_unactualized_session_wrappers.erase(i);
-        wrapper->finalize_before_actualization();
-        return;
+    bool retrigger;
+    {
```

Comment:
This just moves the lock_guard into its own scope, since calling `m_actualize_and_finalize->trigger()` wakes up the event loop thread which will immediately try to acquire `m_mutex` and wait on a cond var - so this should hopefully reduce the likelihood the event loop will have to block for the mutex.